### ISSUE TITLE
error when in loops

### ISSUE THREAD

**Issue by TY-Cheng** [vinecopulib/pyvinecopulib#77]
Hi PyVineCop Master!

Thanks for your kind work. I got this error in loops when I was fitting vine copula based on rolling windows.

RuntimeError: not a valid R-vine array: the order/antidiagonal must contain the numbers 1, ..., d (the number of variables)
<img width="918" alt="Screenshot 2021-05-11 at 6 05 27 AM" src="https://user-images.githubusercontent.com/14033617/117730633-3b169c80-b21f-11eb-9670-5e542fc32f3a.png">

After many iterations (more than 100) I got this error. I checked the input data array, it's okay. When I continue from where it is, the bug is gone.

I prepare the Vinecop object and the FitControlsVinecop object first then refit in every iteration. 

outside loop:
import pyvinecopulib as pvc
iter_VCop = pvc.Vinecop(d=len(seq_asset))
iter_VCop_FC = pvc.FitControlsVinecop(
    selection_criterion='aic', select_trunc_lvl=True, select_threshold=True,
    show_trace=False, num_threads=num_cores
)
in loop:
update df_V_NN
iter_VCop.select(data=df_V_NN, controls=iter_VCop_FC)

Should I __init__ them every time?
Are these objects 'disposable' or can be 'reused'? 

Thx!

**Comment by tvatter** [comment]
Hi,

In theory, your code is correct and the objects can be 'reused' (i.e., you shouldn't need to init them every time). Now, I am not sure what causes the bug in your specific case, and it is a bit tricky to debug without an MWE allowing me to reproduce the issue. 

If you can't share data allowing me to reproduce the issue, I would suggest that inside the loop, you use:

iter_VCop = pvc.Vinecop(data=df_V_NN, controls=iter_VCop_FC)

In other words, the Vinecop class has a constructor taking a dataset + controls as arguments. It should be slightly less efficient, but in your setup (selection_criterion='aic', select_trunc_lvl=True, select_threshold=True), creating the initial model shouldn't be the computational bottleneck anyway.

Hope this helps,
Thibault

**Comment by TY-Cheng** [comment]
Hi Thibault:

thx for your help. 

Sorry, it persists... would it be an out-of-memory problem? was using df_V_NN as a matrix of 600*15 in a loop of 1000 iterations. 
thx!


![image](https://user-images.githubusercontent.com/14033617/118304777-b034eb00-b519-11eb-9344-d368e04aa34a.png)

**Comment by tvatter** [comment]
Did you try what I recommended or can you share a mwe? It's definitely not a memory issue. If you just try to fit the model for the loop index where it fails (i.e., manually), can you reproduce the issue? And if yes, could you extract this subdataset and share it with me in csv without column names? I wouldn't know what the data is about if you are concerned about privacy, but I could try to figure out the issue.
